src/components/FavoritesBtn.tsx

```diff
@@ -6,30 +6,40 @@ import '../styles/FavoritesBtn.scss';
 
 type FavoritesBtnProps = {
   movieId: number;
+  isMovieIdInFavorites?: boolean;
 };
 
-const FavoritesBtn: React.FC<FavoritesBtnProps> = ({ movieId }) => {
+const FavoritesBtn: React.FC<FavoritesBtnProps> = ({
+  movieId,
+  isMovieIdInFavorites
+}) => {
   const [isFavorite, setIsFavorite] = useState(api.isIdInFavorites(movieId));
- 
+
   const heartUnfilledIcon = '/images/favoriteBtn/heartFilled.svg';
   const heartFilledIcon = '/images/favoriteBtn/heartUnfilled.svg';
 
   const handleSwitchFavoriteState = (): void => {
-    isFavorite ? api.setFavoritesId(movieId) : api.deleteFavoritsId(movieId);
-    setIsFavorite(api.isIdInFavorites(movieId));
+    isFavorite ? api.deleteFavoritsId(movieId) : api.setFavoritesId(movieId);
+    setIsFavorite(!isFavorite);
   };
 
   return (
     <button className="favorite-btn" onClick={handleSwitchFavoriteState}>
-      <img
-        className="favorite-icon"
-        src={isFavorite ? heartFilledIcon : heartUnfilledIcon}
-        alt="favorite icon"
-      />
+      {isMovieIdInFavorites ? (
+        <img
+          className="favorite-icon"
+          src={isMovieIdInFavorites ? heartUnfilledIcon : heartFilledIcon}
+          alt="favorite icon"
+        />
+      ) : (
+        <img
+          className="favorite-icon"
+          src={isFavorite ? heartUnfilledIcon : heartFilledIcon}
+          alt="favorite icon"
+        />
+      )}
```

Comment:
cant get this part

Reply:
Here, I was trying to figure out the way how to handle changing a favorite icon state when a user click a 5-star icon (maximum rate). I used a different approach (hopefully now it's correct :) )